### ISSUE TITLE
Should be able to add additional annotations to beyond exclude

### ISSUE THREAD

**Issue by dovidkopel** [benas/random-beans#156]

**Comment by PascalSchumacher** [comment]
Sorry, but I do not understand what this issue is suggesting. Can please you provide a more detailed explanation and an example?

**Comment by benas** [comment]
Thank you for this PR. It would be easier to give a brief description of the issue to the maintainer.

Looking at your changes, I see the idea. Basically, you want to exclude a field if it is annotated with one of the given annotations. Is this correct?

I've got an interesting discussion on this subject [here](https://github.com/benas/random-beans/issues/82#issuecomment-190282286). I invite you to take a quick look. My idea was to add this feature using the `FieldDefinition`, something like:

```java
EnhancedRandomBuilder.aNewEnhancedRandom()
.exclude(field().isAnnotatedWith(MyAnnotation.class).get())
.build();
```

What do you think?

The second option is to use a custom registry. You can simply add a custom registry that looks if the field is annotated with one of the given annotations and return a `SkipRandomizer`, similar to the built-in [AnnotationRandomizerRegistry](https://github.com/benas/random-beans/blob/master/random-beans/src/main/java/io/github/benas/randombeans/randomizers/registry/AnnotationRandomizerRegistry.java#L55). I think this is easier than changing the logic of field exclusion.

**Comment by benas** [comment]
Hi Dovid,

Here is a quick example of how we can achieve the requested feature with a custom registry (option 2 in my previous comment):

```java
@Priority(1)
public class CustomExclusionAnnotationRegistry implements RandomizerRegistry {

    private Annotation[] annotations;

    public CustomExclusionAnnotationRegistry(Annotation... annotations) {
        this.annotations = annotations;
    }
    @Override
    public Randomizer<?> getRandomizer(Field field) {
        for (Annotation annotation : annotations) {
            if (field.isAnnotationPresent(annotation.getClass())) {
                return new SkipRandomizer();
            }
        }
        return null;
    }
}
```

If I want to exclude fields annotated with `@MyAnnotation`, I can simply register this registry:

```java
EnhancedRandomBuilder.aNewEnHancedRandom()
.registerRandomizerRegistry(new CustomExclusionAnnotationRegistry(MyAnnotation.class)
.build();
```

The `CustomExclusionAnnotationRegistry` would be part of the public API of random beans so that users can use it to exclude fields with their custom annotations.

I do believe this solution follows the open closed principle. What do you think?

Kind regards
Mahmoud

**Comment by benas** [comment]
Hi Dovid,

I made a refactoring (see issue #160) to the `FieldDefinition` class in order to be able to write something like:

```java
EnhancedRandom random = aNewEnhancedRandomBuilder()
   .exclude(field().isAnnotatedWith(Myannotation.class).get())
   .build();
```

It's now possible to exclude fields annotated with a custom annotation in a consistent way with other exclusions (by name, by type, etc). This addition is available in version 3.2.0-SNAPSHOT.

It would be great if you can give it a try and give me your feedback.

Kind regards
Mahmoud

**Comment by dovidkopel** [comment]
Looks good, sorry about the delay. Very clean approach. Will try shortly.
